src/Illuminate/Support/Collection.php

```diff
@@ -1350,3 +1369,36 @@ protected function getArrayableItems($items)
         return (array) $items;
     }
 }
+
+class HigherOrderProxy
+{
+    /**
+     * @var Collection
+     */
+    protected $collection;
+
+    /**
+     * @var string
+     */
+    protected $method;
+
+    public function __construct(Collection $collection, $method)
+    {
+        $this->collection = $collection;
+        $this->method = $method;
+    }
+
+    public function __call($name, $arguments)
+    {
+        return $this->collection->{$this->method}(function($value) use ($name, $arguments) {
+            return call_user_func_array([$value, $name], $arguments);
+        });
+    }
+
+    public function __get($name)
+    {
+        return $this->collection->{$this->method}(function($value) use ($name) {
+            return $value->$name;
```

Comment:
Perhaps we shouldn't assume the value will be an object. We also use collections for arrays.